Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h

```diff
@@ -201,6 +201,10 @@ struct CLuaFunctionParserBase
         if constexpr (std::is_same_v<T, CLuaArgument>)
             return iArgument != LUA_TNONE;
 
+        // all color classes are read as 1 number, not 3 different.
+        if constexpr (std::is_base_of_v<SColor, T>)
+            return iArgument == LUA_TNUMBER;
```

Comment:
Should LUA_TSTRING be allowed too? (because in Lua strings autoconvert to numbers.)

(how does the lua parser reject bad numberstrings?)

Reply:
idk if you're asking me. The old one warns if the string isnt convertible to a number. The new one probably hard errors or something.

tbh, not quite sure what happens if i pass in `"NAH"` and it tries to read it as a number, but its a string.

Reply:
```suggestion
        // all color classes are read as 1 number, not 3 different.
        if constexpr (std::is_base_of_v<SColor, T>)
            return lua_isnumber(L, index);
```